xla/service/collective_pipeliner.cc

```diff
@@ -508,10 +511,12 @@ std::optional<std::vector<HloInstruction*>> CollectIndependentOperandChain(
     const bool is_scalar_shaped =
         ShapeUtil::IsEffectiveScalar(chain_instr->shape());
     if (!all_users_in_chain) {
-      if (!loop_invariant_params.contains(chain_instr) && !is_scalar_shaped &&
-          (chain_instr->opcode() != HloOpcode::kGetTupleElement ||
-           chain_instr->operand(0)->opcode() != HloOpcode::kParameter ||
-           !should_allow_loop_variant_parameter_in_chain(chain_instr))) {
+      if ((!loop_invariant_params.contains(chain_instr) && !is_scalar_shaped &&
```

Comment:
Can yo unwrap this expression and give it some local names?

Reply:
i extracted the partial logic out of this entire expression, let me know if i understood your intention correctly.